# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: uriloader/exthandler/nsExternalHelperAppService.cpp
# Commit: f0094c9184c4
# Full Hash: f0094c9184c41675f11efc2d4332ec1a63e570bc
# Author: Gijs Kruitbosch <gijskruitbosch@gmail.com>
# Date: 2019-11-27 09:44:05
# Regressor Bug: 1496380
# File Overlap Count: 1
# Description:
#   Bug 1496380 - stop recursion via the external protocol handler if Firefox is either the default OS handler or a configured external handler, r=mossop
#   
#   This is an initial implementation of this idea that works on mac.
#   I've added a Windows implementation in another commit in this stack. I'll
#   look at a Linux one in a follow-up bug. I do not think we need them in the
# ==============================================================================

diff -r 9dbf3a2c8c37 -r f0094c9184c4 uriloader/exthandler/nsExternalHelperAppService.cpp
--- a/uriloader/exthandler/nsExternalHelperAppService.cpp	Tue Nov 26 18:47:20 2019 +0000
+++ b/uriloader/exthandler/nsExternalHelperAppService.cpp	Tue Nov 26 18:51:04 2019 +0000
@@ -1008,13 +1008,6 @@
                       nsIContentDispatchChooser::REASON_CANNOT_HANDLE);
 }
 
-NS_IMETHODIMP nsExternalHelperAppService::GetApplicationDescription(
-    const nsACString& aScheme, nsAString& _retval) {
-  // this method should only be implemented by each OS specific implementation
-  // of this service.
-  return NS_ERROR_NOT_IMPLEMENTED;
-}
-
 //////////////////////////////////////////////////////////////////////////////////////////////////////
 // Methods related to deleting temporary files on exit
 //////////////////////////////////////////////////////////////////////////////////////////////////////
@@ -1102,13 +1095,6 @@
 }
 
 NS_IMETHODIMP
-nsExternalHelperAppService::GetProtocolHandlerInfoFromOS(
-    const nsACString& aScheme, bool* found, nsIHandlerInfo** aHandlerInfo) {
-  // intended to be implemented by the subclass
-  return NS_ERROR_NOT_IMPLEMENTED;
-}
-
-NS_IMETHODIMP
 nsExternalHelperAppService::SetProtocolHandlerDefaults(
     nsIHandlerInfo* aHandlerInfo, bool aOSHandlerExists) {
   // this type isn't in our database, so we've only got an OS default handler,